sentry_streams/sentry_streams/adapters/arroyo/consumer.py

```diff
@@ -59,9 +60,26 @@ def build_strategy(self, commit: Commit) -> ProcessingStrategy[Any]:
         follow.
         """
 
-        def add_route(message: Message[KafkaPayload]) -> RoutedValue:
-            value = message.payload.value
-            return RoutedValue(route=Route(source=self.source, waypoints=[]), payload=value)
+        def add_route(message: Message[KafkaPayload]) -> Union[FilteredPayload, RoutedValue]:
+            filtered = False
+            if self.header_filter:
+                headers: Headers = message.payload.headers
+                if self.header_filter not in headers:
+                    filtered = True
+
+            if not filtered:
+                value = message.payload.value
+                try:
+                    schema: Codec[Any] = get_codec(self.stream_name)
+                except Exception:
+                    raise ValueError(f"Kafka topic {self.stream_name} has no associated schema")
+
+                return RoutedValue(
+                    route=Route(source=self.source, waypoints=[]),
+                    payload=StreamsMessage(schema=schema, payload=value),
```

Comment:
This identifies the schema of the messages based on the topic it comes from. I chose to do it this way, and wrap it in this `Message` so that we can do flexible parsing/deserialization of messages. This way, instead of baking it into the Source, we can basically do parsing whenever in the pipeline using the `Parser()` step.